packages/payload/src/config/types.ts

```diff
@@ -1294,12 +1255,12 @@ export type EditConfigWithoutRoot = {
    *
    * To override the entire Edit View including all nested views, use the `root` key.
    */
-  api?: Partial<EditViewConfig>
-  default?: Partial<EditViewConfig>
-  livePreview?: Partial<EditViewConfig>
+  api?: EditViewConfig
+  default?: EditViewConfig
+  livePreview?: EditViewConfig
   root?: never
-  version?: Partial<EditViewConfig>
-  versions?: Partial<EditViewConfig>
+  version?: EditViewConfig
+  versions?: EditViewConfig
```

Comment:
If someone defines any of these properties, they will now have to define non-optional subproperties, such as `Component`. Is this intentional?

Reply:
> they will now have to define non-optional subproperties

Given that this is a union type, `Component` wouldn't be required. Regardless, this was unintentional. We probably 
 should, however, make this change in the future, as using `Partial` negates our strong typings here.